### PR TITLE
Add preliminary support for GNU/Hurd

### PR DESCRIPTION
**Pull request by ibuclaw**
Druntime compiles, but has not been thoroughly tested yet.

If this seems small, that would be down to this being only the changes required against the dmd-cxx branch bindings.

Bootstrapping against master will likely reveal more gaps that need filling in as a result of the recent CRuntime_Glibc refactorings.

### CODE REVIEW

File: src/core/sys/linux/elf.d [hunk 1]
```diff
@@ -5,7 +5,15 @@
...
-version (linux):
+// This interface is really a CRuntime_Glibc header, but as the bindings are
+// in a platform-specific module, that would mean this 2500+ line file would
+// have to be copied for each platform that is exclusively glibc.
+// Selectively permit other platforms to import the symbols in this module.
+
+version (linux) version = LinuxOrCompatible;
+version (Hurd)  version = LinuxOrCompatible;
+
+version (LinuxOrCompatible):
```
Comment (ibuclaw) [inline src/core/sys/linux/elf.d:16]: This is the only weird case that needs considering.

Comment (ljmf00) [reply src/core/sys/linux/elf.d:16]: ELF for Linux or Hurd, is exactly the same according to the OS identifier.

Comment (sthibaul) [reply src/core/sys/linux/elf.d:16]: yes, they are the same

**Review by ljmf00 — APPROVED** [review]

File: src/core/time.d [hunk 2]
```diff
@@ -429,6 +439,20 @@ version (Posix)
...
             }
         }
+        else version (Hurd)
+        {
+            import core.sys.hurd.time;
+            with(ClockType) final switch (clockType)
+            {
+            case coarse: return CLOCK_MONOTONIC_COARSE;
+            case normal: return CLOCK_MONOTONIC;
+            case precise: return CLOCK_MONOTONIC;
```
Comment (ljmf00) [inline src/core/time.d:449]: You don't need to repeat this.

Comment (ibuclaw) [reply src/core/time.d:449]: What are you talking about?

Comment (ljmf00) [reply src/core/time.d:449]:
```suggestion
            case normal:
            case precise:
            	return CLOCK_MONOTONIC;
```

**Comment by dkorpel** [comment]
Are you still working on this @ibuclaw ?

**Comment by Geod24** [comment]
Druntime have been merged into DMD. Please re-submit your PR to `dlang/dmd` repository.

**Pull request closed without merging**